decidim-participatory_processes/app/controllers/decidim/participatory_processes/widgets_controller.rb

```diff
@@ -5,13 +5,19 @@ module ParticipatoryProcesses
     class WidgetsController < Decidim::WidgetsController
       helper Decidim::SanitizeHelper
 
+      def show
+        enforce_permission_to :read, :participatory_space, current_participatory_space: model
+
+        super
+      end
+
       private
 
       def model
         return unless params[:participatory_process_slug]
 
-        @model ||= ParticipatoryProcess.where(slug: params[:participatory_process_slug]).or(
-          ParticipatoryProcess.where(id: params[:participatory_process_slug])
+        @model ||= ParticipatoryProcess.published.where(slug: params[:participatory_process_slug]).or(
```

Comment:
Here we have a possible security issue when multi tenancy is involved. Reading the piece of code, it seems that is possible to access the widgets cross organization.

Reply:
Solved with 059fd6067a